recommender/engines/nlp_embedders/handlers.py

```diff
@@ -0,0 +1,153 @@
+        self.field_name = field_name
+        self.accumulator = []
+
+    def __call__(self, obj):
+        self.accumulator.append(self.text2vec(getattr(obj, self.field_name)))
+
+    def get_results(self) -> List[Tensor]:
+        result = [torch.stack(self.accumulator)]
+        self.accumulator = []
+        return result
+
+
+class ListOfStringsHandler(Handler):
+    def __init__(self, field_name, text2vec=None):
+        super().__init__(text2vec)
+        self.field_name = field_name
+        self.accumulator = []
+
+    def __call__(self, obj):
+        self.accumulator.append(self.text2vec(", ".join(getattr(obj, self.field_name))))
+
+    def get_results(self) -> List[Tensor]:
+        result = [torch.stack(self.accumulator)]
+        self.accumulator = []
+        return result
+
+
+class ObjectHandler(Handler):
+    def __init__(self, field_name, subfields_names, text2vec=None) -> None:
+        super().__init__(text2vec)
+        self.field_name = field_name
+        self.subfields_names = subfields_names
+        self.accumulators = self._set_accumulators()
+
+    def __call__(self, obj) -> None:
+        for subfield_name, accumulator in self.accumulators.items():
+            obj = getattr(obj, self.field_name)
+            accumulator.append(self.text2vec(getattr(obj, subfield_name)))
+
+    def _set_accumulators(self):
+        return {subfield_name: [] for subfield_name in self.subfields_names}
+
+    def get_results(self) -> List[Tensor]:
+        # There could be some flattening strategy injection
+        result = [
+            torch.stack(accumulator) for accumulator in self.accumulators.values()
+        ]
+        self.accumulators = self._set_accumulators()
+        return result
+
+    @property
+    def embedding_dim(self):
+        return self.text2vec.embedding_dim * len(self.accumulators.items())
+
+
+class ListOfObjectsHandler(Handler):
+    MAGIC_NUMBER = 512  # refactor
+
+    def __init__(
+        self,
+        field_name,
+        subfields_names,
+        text2vec=None,
+        flattening_strategy: Union[Callable[[Tensor], Tensor], None] = None,
+    ) -> None:
+        super().__init__(text2vec)
+        self.field_name = field_name
+        self.subfields_names = subfields_names
+        self.accumulators = self._set_accumulators()
+        self.flattening_strategy = flattening_strategy or self._flattening_strategy
+
+    def __call__(self, obj) -> None:
+        subobjects = getattr(obj, self.field_name)
+        for subfield_name, accumulator in self.accumulators.items():
+            if subobjects:
+                seq = []
+                for subobj in subobjects:
+                    subfield_val = getattr(subobj, subfield_name)
+                    if subfield_val is None:
+                        subfield_val = ""
+                    seq.append(self.text2vec(subfield_val))
+                tensor = torch.stack(seq)
+            else:
+                tensor = torch.zeros(1, self.MAGIC_NUMBER)
```

Comment:
This is part of the same story as in the case of hard-cut workaround (explained in one of the previous comments and answers) - it will be refactored in the next NLP-related PR